src/Purebred/Storage/Notmuch.hs

```diff
@@ -208,16 +208,17 @@ messageToMail m = do
       <*> Notmuch.messageId m
 
 -- | Returns the notmuch database path by executing 'notmuch config
--- get database.path' in a separate process
+-- get database.path' in a separate process.  If the process terminates
+-- abnormally, returns an empty string.
 --
 getDatabasePath :: IO FilePath
 getDatabasePath = do
   let cmd = "notmuch"
   let args = ["config", "get", "database.path"]
-  (exitc, stdout, err) <- readProcess $ proc cmd args
-  case exitc of
-    ExitFailure _ -> error (untaint decode err)
-    ExitSuccess -> pure (filter (/= '\n') (untaint decode stdout))
+  (exitc, stdout, _err) <- readProcess $ proc cmd args
+  pure $ case exitc of
+    ExitFailure _ -> ""
```

Comment:
We are now unconditionally executing the function, whereas previously the user had an opportunity to substitute a different "get database path" function (e.g. to supply a constant, alternative database location).  Now, some older versions of notmuch (including v0.26, the version in CI), when executing @notmuch config get ...@, exit abnormally if it cannot find a configuration file, whereas newer versions exit cleanly without producing any output.

To handle all these discrepancies, I thought it was best to NOT fail in this function.  The use can still override the database location, and purebred will fail elsewhere (hopefully with a reasonable error message - perhaps I should check) if the database location is invalid.

Reply:
To add some more context: notmuch >= v0.33 has much richer heuristics for locating a database and doesn't need the `database.path` config setting unless the database is in an unusual location.  In response to these improvements, we can simplify purebred's behaviour substantially.  Except we can't just yet, because older notmuch versions in the CI environment are holding us back.

The main takeaway is that hacks like ^^^ are tolerable, because they'll eventually disappear.

Reply:
Aaah I see. Fair enough.